### PR TITLE
Ports Yogstation's version of CQC, adds it back to traitor uplink.

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do
Ports Yogstation's version of CQC, much more focused on dealing stamina damage and non-lethal takedowns. Adds it back into traitor uplink.
## Why It's Good For The Game

<details><summary>For starters, this is Yog's CQC</summary>

>All of your unarmed attacks deal stamina damage instead of your normal physical damage type

>Disarm Intent Has a chance to disarm the opponent's main hand, and immediately pick up the item if successful
>Grab Intent Will stun opponents for a short second, allowing you to quickly increase the strength of your grabs
>Harm Intent Will deal a competitive amount of stamina damage, and hitting a standing opponent while you are prone will both >knock them down and stand you up

>Slam: Grab Harm. Slam opponent into the ground, knocking them down and dealing decent stamina damage.
>CQC Kick: Disarm Harm. Knocks opponent away and slows them. Deals heavy stamina damage to prone opponents, as well as muting them for a short time.
>Restrain: Grab Grab. Locks opponents into a restraining position, making your grab harder to break out of, disarm to begin a chokehold which deal gradual oxygen damage until the opponent is unconscious, with the damage increasing based on their stamina damage. Failing to complete the chokehold will weaken and possibly break your grab.
>Pressure: Disarm Grab. Disables the targetted limb or a random limb if the head or chest are targetted, as well as forcing the target to drop
>Consecutive CQC: Harm Harm Harm Harm Harm. Offensive move, deals bonus stamina damage and knocking down on the last hit.
>In addition, by having your throw mode on when being attacked, you enter an active defense mode where you have a chance to counter attacks done to you.
</details>

CQC was a very fun martial art that myself and others loved to use. However in its current state, it was very fairly moved to be for nukies only, making it so sleeping carp was the only martial art with combos available to most traitors. This version of CQC rebalances it to make it much more fair to play against, most noteably by changing chokehold to take some time to knock the target out, while also giving it a strong niche of being a much more non-lethal option to traitoring. However even this will need re-balancing as well, which I am very open to feedback for.

<details><summary>Here are my current proposed changes</summary>

Disarm Intent: Instead just have shoving someone deal stamina damage, to avoid any RNG in combat

Grab Intent: Keep the small stun maybe? I'd like to have some sort of special effect, and instant aggressive grabs feel too strong.

Pressure: Have this be the combo for stealing someone's item.

Consecutive CQC: Keep our combo of disarm, disarm, harm, and have it deal brute damage, so there's still an option of dealing damage without a weapon.

Throw Mode Block Chance: Honestly, I have no idea what to do about this, and would love suggestions. RNG combat can feel really unfair, so changing this is a priority.

</details>

On Discord, Charlie mentioned that he'd rather tweak our existing martial arts instead of adding new ones, and he expressed interest in this version of CQC when I initially pitched it. However because its not a concrete answer, and Charlie isn't the only person on the development/design/balance team, I decided to keep this as WIP so I don't have to do the hard part until I get full approval.

Todo List:
- [x] Change most instances of brute damage to stamina damage.
- [x] Change combos to be Yog's combos
- [x] Figure out what to do about RNG block/counter
- [x] Change chokehold into Yog's chokehold
- [x] Add it into traitor uplink
- [x] Fix the admin attack log messages to be more accurate
- [x] Generally go back and make sure everything does what it's supposed to
- [x] Balance the values/effects

## Testing

Spawned in and gave myself CQC, robusted some skrells.
I'll update this as I test more and more.

## Changelog
:cl:
add: A tweaked version of Yogstation's CQC
del: Our old CQC
tweak: CQC now available in traitor uplink,
tweak: CQC is now 8 TC for nuclear agents
/:cl:

### CODE REVIEW

**Comment by FloFluoro** [comment]
This PR would also progress #18179. In fact, it's one of the last things on the list that hasn't been done yet.

**Comment by AzuleUtama** [comment]
I'm a bit mixed on re-adding CQC to traitors once again - it's been in and out of the uplink multiple times now and I think it would make more sense to keep it exclusive to nukies (and the 'Bond' syndicate bundle) while fixing the instant stun mechanics it still has. Traitors have Carp Fu exclusively and both antags have access to the Desword, Chainsaw and Krav Maga gloves - so Nuke Ops having an exclusive and powerful martial art of their own would make CQC feel more rewarding when you're lucky enough to use it.

File: code/modules/martial_arts/combos/cqc/kick.dm [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 		target.throw_at(throw_target, 1, 14, user)
-		target.apply_damage(10, BRUTE)
+		target.apply_damage(15, STAMINA)
+		target.Slowed(5 SECONDS)
 		add_attack_logs(user, target, "Melee attacked with martial-art [src] : Kick", ATKLOG_ALL)
 		. = MARTIAL_COMBO_DONE
-	if(target.IsWeakened() && !target.stat)
-		target.visible_message("<span class='warning'>[user] kicks [target]'s head, knocking [target.p_them()] out!</span>", \
-					  		"<span class='userdanger'>[user] kicks your head, knocking you out!</span>")
+	if(IS_HORIZONTAL(target) && !target.stat)
```
Comment (hal9000PR) [inline code/modules/martial_arts/combos/cqc/kick.dm:18]: no need to check for stat if you are checking for horizontal

File: code/modules/martial_arts/cqc.dm [hunk 2]
```diff
@@ -20,12 +21,32 @@
...
+/datum/martial_art/cqc/proc/drop_chokehold()
+	chokehold_active = FALSE
+
+/datum/martial_art/cqc/proc/start_chokehold(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	D.visible_message("<span class='danger'>[A] puts [D] into a chokehold!</span>", \
+						"<span class='userdanger'>[A] puts you into a chokehold!</span>")
+	add_attack_logs(A, D, "Put into a chokehold with martial-art [src]", ATKLOG_ALL)
+	chokehold_active = TRUE
+	while(do_mob(A, D, 2 SECONDS) && chokehold_active)
+		var/damage_multiplier = 1 + A.getStaminaLoss()/100 //The chokehold is more effective the more tired the target is.
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:33]:
```suggestion
		var/damage_multiplier = 1 + A.getStaminaLoss() / 100 //The chokehold is more effective the more tired the target is.
```
spacing, also does this need to be recalculated each loop?

File: code/modules/martial_arts/cqc.dm [hunk 3]
```diff
@@ -38,7 +59,7 @@
 	if(D.IsWeakened() || IS_HORIZONTAL(D))
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:59]: no need to check for weakened anymore

File: code/modules/martial_arts/cqc.dm [hunk 4]
```diff
@@ -38,7 +59,7 @@
...
 		bonus_damage += 5
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:60]: might want to increase this if its just stam damage now

File: code/modules/martial_arts/cqc.dm [hunk 5]
```diff
@@ -50,42 +71,32 @@
 		D.visible_message("<span class='warning'>[A] leg sweeps [D]!", \
 							"<span class='userdanger'>[A] leg sweeps you!</span>")
 		playsound(get_turf(A), 'sound/effects/hit_kick.ogg', 50, 1, -1)
-		D.apply_damage(10, BRUTE)
-		D.Weaken(6 SECONDS)
+		D.KnockDown(3 SECONDS)
+		A.SetKnockDown(0 SECONDS) //Quickly get up like the cool dude you are.
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:75]: might want to call stand_up() if you want them to stand up instantly

File: code/modules/martial_arts/cqc.dm [hunk 6]
```diff
@@ -50,42 +71,32 @@
...
-		if(!D.stat || !D.IsWeakened() || !restraining)
-			I = D.get_active_hand()
-			D.visible_message("<span class='warning'>[A] strikes [D]'s jaw with their hand!</span>", \
-								"<span class='userdanger'>[A] strikes your jaw, disorienting you!</span>")
-			playsound(get_turf(D), 'sound/weapons/cqchit1.ogg', 50, TRUE, -1)
-			if(D.unEquip(I) && !(QDELETED(I) || (I.flags & ABSTRACT)))
-				A.put_in_hands(I)
-			D.Jitter(4 SECONDS)
-			D.apply_damage(5, BRUTE)
+	if(!D.stat || !D.IsWeakened() || !restraining)
```
Comment (hal9000PR) [inline code/modules/martial_arts/cqc.dm:89]: are you trying to check if they are stood up?

**Review by hal9000PR — CHANGES_REQUESTED** [review]

**Comment by Edan52** [comment]
Hi, what did you do with the RNG?

File: code/datums/uplink_item.dm [hunk 7]
```diff
@@ -891,6 +891,12 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/CQC_manual
 	cost = 13
 	cant_discount = TRUE
+	excludefrom = list(UPLINK_TYPE_NUCLEAR, UPLINK_TYPE_SST)
+
+/datum/uplink_item/stealthy_weapons/cqc/nuke
+	reference = "NCQC"
+	cost = 8
+	excludefrom = list()
 	uplinktypes = list(UPLINK_TYPE_NUCLEAR, UPLINK_TYPE_SST)
```
Comment (hal9000PR) [inline code/datums/uplink_item.dm:900]:
```suggestion
	excludefrom = list(UPLINK_TYPE_TRAITOR, UPLINK_TYPE_SIT)
```
I think this looks better, and it prevents admin uplinks getting 2 CQC options

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**